packages/web/src/react.test.tsx

```diff
@@ -46,4 +46,83 @@ describe('<Analytics />', () => {
       expect(script).toHaveAttribute('defer');
     });
   });
+
+  describe('track  custom events', () => {
+    beforeEach(() => {
+      // reset the internal queue before every test
+      window.vaq = [];
+    });
+    describe('queue custom events', () => {
+      it('event name only', () => {
+        render(<Analytics mode="production" />);
+
+        track('my event');
+
+        expect(window.vaq).toBeDefined();
+
+        if (!window.vaq) throw new Error('window.vaq is not defined');
+
+        expect(window.vaq[0]).toEqual([
+          'track',
+          {
+            name: 'my event',
+          },
+        ]);
+      });
+      it('with custom data', () => {
+        render(<Analytics mode="production" />);
+
+        track('custom event', {
+          string: 'string',
+          number: 1,
+        });
+
+        expect(window.vaq).toBeDefined();
+
+        if (!window.vaq) throw new Error('window.vaq is not defined');
+
+        expect(window.vaq[0]).toEqual([
+          'track',
+          {
+            name: 'custom event',
+            data: {
+              string: 'string',
+              number: 1,
+            },
+          },
+        ]);
+      });
+
+      it('strip nested object', () => {
+        process.env = {
+          NODE_ENV: 'production',
+        };
```

Comment:
That's not working as it has no effect – why do we need it?
We introduced the `mode` property for this reason.

Reply:
I just saw that I used `isDevelopment` in the `track` function, thats why.
We need to think about a solution on how I can access `mode` without manually passing it to `track`.

Reply:
For that we have `getMode`. It's also using `isDevelopment` under the hood

Reply:
I believe we have to store the `mode` on a global level, so subsequent calls such as `track` can read that value that was provided on `inject`. That should allow us to have `track` and other functions without the additional effort of providing a `mode` property individually.

1. `inject` is called with `mode`
2. we store the `mode` on `va` globally
3. `track` reads mode from global object as it was already set by `inject`

Reply:
@chriswdmr `inject` now sets `window.vam` to the mode that is set. Track later reads this to know if its prod or not!